tests/testthat/test-audit.R

```diff
@@ -46,3 +12,45 @@ without_internet({
     )
   })
 })
+
+with_mock_api({
+  test_that("audit_r_versions does not error", {
+    skip_if_not_installed("ggplot2")
+    skip_if_not_installed("gridExtra")
+    con <- Connect$new(server = "https://connect.example", api_key = "fake")
+    content <- get_content(con)
+
+    expect_s3_class(audit_r_versions(content), "gtable")
+  })
+
+  test_that("audit_access_open", {
+    con <- Connect$new(server = "https://connect.example", api_key = "fake")
+    content <- get_content(con)
+
+    expect_identical(audit_access_open(content, "all"), "My-Streamlit-app")
+    expect_identical(
+      audit_access_open(content, "logged_in"),
+      c("team-admin-dashboard", "Performance-Data-1671216053560")
+    )
+  })
+
+  test_that("audit_runas", {
+    con <- Connect$new(server = "https://connect.example", api_key = "fake")
+    content <- get_content(con)
+    expect_equal(
+      audit_runas(content),
+      tibble::tibble(
+        app_name = "My-Streamlit-app",
+        run_as_user = "current user"
+      )
+    )
+  })
+
+  test_that("vanity_is_available", {
+    con <- Connect$new(server = "https://connect.example", api_key = "fake")
+    expect_true(vanity_is_available(con, "not-currently/used"))
+    expect_false(vanity_is_available(con, "/streamlit/my-app/"))
+    expect_false(vanity_is_available(con, "streamlit/my-app"))
+    expect_false(vanity_is_available(con, "https://connect.example/streamlit/my-app"))
```

Comment:
🤯 I don't think I had ever thought that you could even make a "hierarchy" of vanities by `reports/report_one`, `reports/report_two`, and then possibly putting up a thing with connect widgets at `reports` 🤯

Reply:
I know  ^^^ isn't what you're testing but seeing it here made me realize